docs/content/reference/file-formats.md

```diff
@@ -1,20 +1,238 @@
+      - upgrade
+  - name: connstr
+    description: MyApp database connnection string
+    type: string
+    env: MYAPP_CONNECTION_STRING
+    sensitive: true
+    source:
+      dependency: mysql
+      output: admin-connstr
+    applyTo:
+      - install
+      - status
+  - name: release-name
+    type: string
+    env: RELEASE
+    default: myapp
+
+state:
+  - name: tfstate
+    description: Store terraform state with the bundle instead of a remote backend
+    path: terraform/terraform.tfstate
+
+dependencies:
+  - name: mysql
+    reference: getporter/mysql:v0.1.1
+    parameters:
+      database: myapp
+
+outputs:
+  - name: app-token
+    description: Access token for MyApp
+    type: file
+    path: /cnab/app/myapp_token
+    sensitive: true
+    applyTo:
+      - install
+  - name: ip-address
+    type: string
+
+images:
+  myapp:
+    repository: example/myapp
+    digest: sha256:568461508c8d220742add8abd226b33534d4269868df4b3178fae1cba3818a6e
+
+install:
+  - helm3:
+      description: "Install MyApp"
+      name: "{{ bundle.parameters.release }}"
+      chart: ./charts/myapp
+      replace: true
+      set:
+        image.repository: "{{ bundle.images.myapp.repository }}"
+        image.digest: "{{ bundle.images.myapp.digest }}"
+
+upgrade:
+  - helm3:
+      name: "{{ bundle.parameters.release }}"
+      chart: ./charts/myapp
+      set:
+        image.repository: "{{ bundle.images.myapp.repository }}"
+        image.digest: "{{ bundle.images.myapp.digest }}"
+
+uninstall:
+  - helm3:
+      purge: true
+      releases:
+        - "{{ bundle.parameters.release }}"  
+
+poke:
+  - exec:
+      command: ./poke-myapp.sh
+
+customActions:
+  status:
+    description: See what's up in there
+    modifies: false
+    stateless: true
+
+status:
+  - exec:
+      command: ./status.sh
+```
+
+| Field                            | Required | Description                                                                                                                                    |
+|----------------------------------|----------|------------------------------------------------------------------------------------------------------------------------------------------------|
+| schemaType                       | false    | The type of document. This isn't used by Porter but is included when Porter outputs the file, so that editors can determine the resource type. |
+| schemaVersion                    | true     | The version of the Bundle schema used in this file.                                                                                            |
+| name                             | true     | The name of the bundle.                                                                                                                        |
+| version                          | false    | The version of the bundle, must adher to [semver v2].                                                                                          |
```

Comment:
Should we mention what value will be used for the bundle's tag if version isn't provided?  Just tested and it appears to be a SHA of some sort (which isn't in itself semver, right?)

Reply:
I fixed the docs to indicate that version is required, though people can set it programmatically during build thanks to the feature you added a while back. So it's not possible to avoid setting version.

The bundle still uses a tag defaulted using the version unless you set it with the reference field, or --tag when publishing.

What you are probably noticing is Yingrong's recent fix for how we first push the invocation image (which used to be named BUNDLENAME-installer), retrieved its repository digest and then patched the bundle.json before publishing the bundle. We now push the invocation image to a tag (a hash) in the same repository as the bundle. We still do the same trick to update the repository digest in the bundle before publishing. The difference is that we don't create additional repositories that are unused (i.e. the -installer image isn't actually used by the bundle, we use the one in BUNDLENAME@sha256:abc123). This makes it more clear to people that we are generating stuff that they shouldn't try to reference by name and that the image is distributed with the bundle.

Untagging so that we don't leave stuff around to confuse people isn't supported yet but is on the backlog.